### PR TITLE
Query route #56 #57:

### PR DESCRIPTION
**Pull request by mantagen**
- creating  route
- creating validation middleware using 'joi'
- creating error handling middleware preferring 'boom' error objects

### CODE REVIEW

File: server/src/middleware/validation.middleware.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import joi from 'joi';
+import boom from 'boom';
+
+const validate = (dataKey, schema) => async (req, res, next) => {
+  const data = req[dataKey];
```
Comment (Jbarget) [inline server/src/middleware/validation.middleware.js]: i dont think we need this.
If you look at what we did on authored it meant we didnt need to specify a key

https://github.com/authored/user-service/blob/develop/server/src/app/routes/index.js#L78

https://github.com/authored/user-service/blob/develop/server/src/app/schema/create-publication-user.js

File: client/src/Views/Home.elm [hunk 2]
```diff
@@ -40,8 +40,23 @@ inputClass =
...
 
-submitButton : Html msg
-submitButton =
-    div [ class "di pa2 absolute right--2 top-0 h3 w3" ]
-        [ button [ class "br-100 bg-blue white f5 metro bn h-100 w-100", value "submit" ] [ text "Go" ]
-        ]
+submitButton : Model -> Html msg
+submitButton model =
+    let
+        countries =
```
Comment (Jbarget) [inline client/src/Views/Home.elm]: country singluar for the moment?

**Comment by Jbarget** [comment]
If i fill in the 3 input boxes and press go i get a 404 on the page, am i supposed to?

screen shot 2018-06-11 at 18 22 13

File: client/src/Views/Home.elm (hunk 2 above)
Comment (mantagen) [reply client/src/Views/Home.elm]: I think countries (with a first pass max 1) makes more sense

**Pull request merged**